Makefile

```diff
@@ -1148,12 +1151,11 @@ $(GINKGO): # Build ginkgo from tools folder.
 ## --------------------------------------
 DOCKER_TEMPLATES := test/e2e/data/infrastructure-docker
 
-.PHONY: generate-e2e-templates
-generate-e2e-templates: $(KUSTOMIZE) generate-e2e-templates-v1alpha3 generate-e2e-templates-v1alpha4 generate-e2e-templates-v1.2 generate-e2e-templates-v1beta1 ## Generate cluster templates for all versions
-
+.PHONY: generate-e2e-templates-v1alpha3
```

Comment:
Should we move all generate targets in the generate section? Seems inconsistent to me to just move one of them

Reply:
see https://github.com/kubernetes-sigs/cluster-api/pull/7329#discussion_r986835619
fine with both approaches for me

Reply:
I would prefer keeping them in one block (and in generate it makes more sense)

Reply:
kk , sooner or later we should break down our make file in smaller mk files and import them

Reply:
I like what the OpenStack provider has done with it's Makefiles. 

For example:
https://github.com/kubernetes-sigs/cluster-api-provider-openstack/blob/main/common.mk
https://github.com/kubernetes-sigs/cluster-api-provider-openstack/blob/main/versions.mk

I would be willing to make a draft for a similar change once all the other Makefile changes are merged.